app/feature/travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/AddCoInsured.kt

```diff
@@ -0,0 +1,130 @@
+package com.hedvig.android.feature.travelcertificate.ui
+
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.LargeContainedButton
+import com.hedvig.android.core.designsystem.component.textfield.HedvigTextField
+import com.hedvig.android.core.designsystem.material3.squircle
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.ValidatedInput
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.feature.travelcertificate.TravelCertificateUiState
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateResult
+import hedvig.resources.R
+import kotlinx.datetime.LocalDate
+
+@Composable
+fun AddCoInsured(
+  selectedCoInsuredId: String?,
+  uiState: TravelCertificateUiState,
+  onSsnChanged: (String) -> Unit,
+  onNameChanged: (String) -> Unit,
+  onRemoveClicked: (String) -> Unit,
+  onSaveClicked: () -> Unit,
+) {
+
+  val selectedCoInsured = uiState.coInsured.input.firstOrNull { it.id == selectedCoInsuredId }
+
+  HedvigScaffold(
+    navigateUp = {},
+    modifier = Modifier.clearFocusOnTap(),
+  ) {
+    Spacer(modifier = Modifier.height(48.dp))
+    Text(
+      text = "Add co-insured",
+      style = MaterialTheme.typography.headlineSmall,
+      textAlign = TextAlign.Center,
+      modifier = Modifier.fillMaxWidth(),
+    )
+    Spacer(modifier = Modifier.height(64.dp))
+    HedvigTextField(
+      value = selectedCoInsured?.name ?: "",
+      onValueChange = { onNameChanged(it) },
+      errorText = "",
```

Comment:
If error text is null, that entire section won't render, but if it's an empty string, the text that would show below the text field will be rendered, and it will add some weird spacing in this case, but simply won't show anything. This is handled here https://github.com/HedvigInsurance/android/blob/93966a0194e063771c25701f721ea26c42ab71b3/app/core/design-system/src/main/kotlin/com/hedvig/android/core/designsystem/component/textfield/HedvigTextField.kt#L106-L121

If we have no error, just pass null instead so that this section isn't rendered in the first place. Same for the other HedvigTextFields in this screen.

Reply:
Fixed in a PR on top of this